src/components/datepicker/DatePicker.js

```diff
@@ -43,6 +43,16 @@ class DatePicker extends PureComponent {
     this.setState({ selectedMonth });
   };
 
+  getMonthPickerSize = () => {
+    const { showWeekNumbers, size } = this.props;
+    const monthPickerSizeByDatePickerSize = {
+      small: 'smallest',
+      medium: showWeekNumbers ? 'medium' : 'small',
+      large: 'large',
+    };
+    return monthPickerSizeByDatePickerSize[size];
+  };
+
```

Comment:
PP: use `useMemo` for this to avoid recalculating on every render, especially since the size won't really change

Reply:
This is a class component though, so do you mean a general memoization function (like from lodash or something)?

Reply:
Ah no, didn't realise it was a class component, you can leave it as is then